acceptance/tests/connect/connect_proxy_lifecycle_test.go

```diff
@@ -0,0 +1,218 @@
+				Secure:      testCfg.secure,
+				ReleaseName: releaseName,
+				Ctx:         ctx,
+				Cfg:         cfg,
+				HelmValues:  testCfg.helmValues,
+			}
+
+			connHelper.Setup(t)
+			connHelper.Install(t)
+			connHelper.DeployClientAndServer(t)
+
+			// TODO: should this move into connhelper.DeployClientAndServer?
+			logger.Log(t, "waiting for static-client and static-server to be registered with Consul")
+			retry.Run(t, func(r *retry.R) {
+				for _, name := range []string{
+					"static-client",
+					"static-client-sidecar-proxy",
+					"static-server",
+					"static-server-sidecar-proxy",
+				} {
+					logger.Logf(t, "checking for %s service in Consul catalog", name)
+					instances, _, err := connHelper.ConsulClient.Catalog().Service(name, "", nil)
+					r.Check(err)
+
+					if len(instances) != 1 {
+						r.Errorf("expected 1 instance of %s", name)
+					}
+				}
+			})
+
+			if testCfg.secure {
+				connHelper.TestConnectionFailureWithoutIntention(t)
+				connHelper.CreateIntention(t)
+			}
+
+			connHelper.TestConnectionSuccess(t)
+
+			// Get static-client pod name
+			// TODO: is this necessary?
+			ns := ctx.KubectlOptions(t).Namespace
+			pods, err := ctx.KubernetesClient(t).CoreV1().Pods(ns).List(
+				context.Background(),
+				metav1.ListOptions{
+					LabelSelector: "app=static-client",
+				},
+			)
+			require.NoError(t, err)
+			require.Len(t, pods.Items, 1)
+			clientPodName := pods.Items[0].Name
+
+			var terminationGracePeriod int64 = 60
+			logger.Logf(t, "killing the %q pod with %dseconds termination grace period", clientPodName, terminationGracePeriod)
+			err = ctx.KubernetesClient(t).CoreV1().Pods(ns).Delete(context.Background(), clientPodName, metav1.DeleteOptions{GracePeriodSeconds: &terminationGracePeriod})
+			require.NoError(t, err)
+
+			// Exec into terminating pod, not just any static-client pod
+			args := []string{"exec", clientPodName, "-c", connhelper.StaticClientName, "--", "curl", "-vvvsSf"}
+
+			if cfg.EnableTransparentProxy {
+				args = append(args, "http://static-server")
+			} else {
+				args = append(args, "http://localhost:1234")
+			}
+
+			if gracePeriodSeconds > 0 {
+				// Ensure outbound requests are still successful during grace
+				// period.
+				retry.RunWith(&retry.Timer{Timeout: time.Duration(gracePeriodSeconds) * time.Second, Wait: 2 * time.Second}, t, func(r *retry.R) {
+					output, err := k8s.RunKubectlAndGetOutputE(t, ctx.KubectlOptions(t), args...)
+					require.NoError(r, err)
+					require.Condition(r, func() bool {
+						exists := false
+						if strings.Contains(output, "curl: (7) Failed to connect") {
+							exists = true
+						}
+						return !exists
+					})
+				})
+
+				// TODO: If listener draining is enabled, ensure inbound
+				// requests are rejected during grace period.
+				// connHelper.TestConnectionSuccess(t)
+
+				// TODO: Test that outbound requests start failing after
+				// shutdown grace period has elapsed but before pod terminates.
```

Comment:
Are these TODOs in scope?

Reply:
Yes, they are out of scope. I will remove them to keep things clean.